akka-http-core/src/main/java/akka/http/javadsl/model/Query.java

```diff
@@ -7,6 +7,7 @@
 import akka.http.impl.model.JavaQuery;
 import akka.http.impl.model.UriJavaAccessor;
 import akka.http.scaladsl.model.*;
+import akka.http.javadsl.model.HttpCharset;
```

Comment:
This is interesting - before adding this import Scala 3 incorrectly resolved javadsl.model.HttpCharset as scaladsl.model.HttpCharset, causing problems in classes that inherit Query.

Reply:
That sounds like a bug in the Java support of Scala 3, doesn't it? Wouldn't it be a problem if javac and scalac disagree about how names are resolved in Java source files?

Reply:
I agree, It also seems like the root of the previous problems with scaladsl and javadsl type inference. Before this I couldn't find a minimization for that issue, but now it seems clear. I'll try making one and file an issue (hopefully also today).